### ISSUE TITLE
Pyright fails to determine the variance of a type variable in a `typing.Protocol`

### ISSUE THREAD

**Issue by LeeeeT** [microsoft/pyright#4674]
For the following code Pyright incorrectly suggests that I change the variance of `Foo` to covariant.

```python
from typing import Generic, Protocol, TypeVar

T_contra = TypeVar("T_contra", contravariant=True)

class Contra(Generic[T_contra]):
    ...

class Foo(Protocol[T_contra]):
    def f(self) -> Contra[T_contra]:
        ...
```

```
file.py:8:7 - error: Type variable "T_contra" used in generic protocol "Foo" should be covariant
```

pyright 1.1.295

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.296, which I just published. It will also be included in a future release of pylance.
